packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -426,6 +426,67 @@ export class ResourceService {
     await this.deleteResourceGitRepository(resource);
   }
 
+  async deleteProjectResources(args: FindOneArgs): Promise<Resource[]> {
+    let deletedServiceCount = 0;
+    const { resources, workspace } = await this.projectService.findUnique({
+      ...args,
+      include: {
+        workspace: true,
+        resources: {
+          where: {
+            deletedAt: null,
+          },
+          include: {
+            gitRepository: true,
+          },
+        },
+      },
+    } as FindOneArgs);
+
+    for (const resource of resources) {
+      if (isEmpty(resource)) {
+        throw new Error(INVALID_RESOURCE_ID);
+      }
+
+      if (resource.resourceType === EnumResourceType.Service) {
+        deletedServiceCount--;
+      }
+    }
+
+    const deleteResources = resources.map((resource) =>
+      this.prisma.resource.update({
+        where: { id: resource.id },
+        data: {
+          name: prepareDeletedItemName(resource.name, resource.id),
+          deletedAt: new Date(),
+          gitRepository: {
+            disconnect: true,
+          },
+        },
+      })
+    );
+    const deletedResources = await this.prisma.$transaction(deleteResources);
+
+    if (this.isBillingEnabled) {
+      await this.billingService.reportUsage(
+        workspace.id,
+        BillingFeature.Services,
+        deletedServiceCount
+      );
```

Comment:
we may need to keep this code anyway even if we not actually deleting the resources, so maybe you could move it to the "deleteProject" function